snapcraft/models/project.py

```diff
@@ -2199,6 +2156,123 @@ def get_partitions(self) -> list[str] | None:
         return _get_partitions_from_components(self.components)
 
 
+class BaseEnum(Enum):
+    BARE = "bare"
+    CORE22 = "core22"
+    CORE24 = "core24"
+    UNIMPLEMENTED = "UNIMPLEMENTED"
+
+    @classmethod
+    def _missing_(cls, value: Any):
+        return cls.UNIMPLEMENTED
+
+
+class BuildBaseEnum(Enum):
+    CORE22 = "core22"
+    CORE24 = "core24"
+    UNIMPLEMENTED = "UNIMPLEMENTED"
+
+    @classmethod
+    def _missing_(cls, value: Any):
+        return cls.UNIMPLEMENTED
+
+
+def _custom_error(error_msg: str):
+    def _validator(v: Any, next_: Any, ctx: pydantic.ValidationInfo):
+        try:
+            return next_(v, ctx)
+        except Exception as exc:
+            raise ValueError(error_msg) from exc
+
+    return pydantic.WrapValidator(_validator)
+
+
+class BaseProject(Project): ...
+
+
+class BareProject(BaseProject): ...
```

Comment:
I'm not sure about this. A bare project should be defined by the bare core22 and bare core24 classes.